docs/user-guide/security.md

```diff
@@ -292,6 +295,293 @@ For example, when using Docker to deploy system containers, the
+unmount it without problem. The new mount can even be stacked on top of an
+immutable mount, in which case it simply "hides" the immutable mount underneath
+(but does not change or remove it, meaning that container isolation remains
+untouched).
+
+### Inner Container Mounts
+
+When launching containers inside a system container (e.g., by installing Docker
+inside the system container), the inner container manager (e.g., Docker) will
+setup several mounts for the inner containers.
+
+This works perfectly fine, as those mounts typically fall under the "other
+mounts" category described in the prior section. However, if any of those mounts
+is associated with an immutable mount of the outer system container, then the
+immutable mount restrictions above apply.
+
+For example, let's launch a system container that comes with Docker inside. We
+will mount as "read-only" a host volume `myvol` into the system container's
+`/mnt/myvol` directory:
+
+```
+$ docker run --runtime=sysbox-runc -it --rm --hostname=syscont -v myvol:/mnt/myvol:ro nestybox/alpine-docker
+```
+
+Inside the system container, let's start Docker:
+
+```
+/ # dockerd > /var/log/dockerd.log 2>&1 &
+```
+
+Now let's launch an inner privileged container, and mount the system container's
+`/mnt/myvol` mount (which is immutable since it was setup at system container
+creation time) into the inner container:
+
+```
+/ # docker run -it --rm --privileged --hostname=inner -v /mnt/myvol:/mnt/myvol ubuntu
+```
+
+Inside the inner privileged container, we can use the mount command. Let's try
+to remount `/mnt/myvol` to read-write:
+
+```
+root@inner:/# mount -o remount,rw,bind /mnt/myvol
+mount: /mnt/myvol: permission denied.
+```
+
+As expected, this failed because the inner container's `/mnt/myvol` is
+technically a bind-mount of `/mnt/myvol` in the system container, and the latter
+is an immutable read-only mount of the system container, so it can't be
+remounted read-write.
+
+
+For the reasons described above, initial mount immutability is a key security
+feature of Sysbox. It enables system container processes to be properly
+isolated from the host, while still giving these processes full access to
+perform other types of mounts, remounts, and unmounts inside the container.
+
+As a point of comparison, other container runtimes either restrict mounts
+completely (by blocking the `mount` and `umount` system calls via the Linux
+seccomp mechanism) which prevents several system-level programs from working
+properly inside the container, or alternatively allow all mount, remount, and
+unmount operations inside the container (e.g., as in Docker privileged
+containers), creating a security weakness that can used to break container
+isolation.
+
+In contrast, Sysbox offers a more nuanced approach, in which the `mount` and
+`umount` system calls are allowed inside the container, but are restricted
+when applied to the container's initial mounts.
```

Comment:
"... to the container's initial mounts (which necessarily correspond to host resources)".

Reply:
Not true; the initial mounts could be tmpfs mounts or other mounts which do not correspond to host resources.

Reply:
True. Disregard.